apps/weather/clkinfo.js

```diff
@@ -14,33 +15,50 @@
         weather.wind = Math.round(weather.wind[1]) + "kph";
     }
 
+    function weatherIcon(code) {
+      var ovr = Graphics.createArrayBuffer(24,24,1,{msb:true});
+      require("weather").drawIcon({code:code},12,12,12,ovr);
+      var img = ovr.asImage();
+      img.transparent = 0;
+      //for (var i=0;i<img.buffer.length;i++) img.buffer[i]^=255;
+      return img;
+      //g.setColor("#0ff").drawImage(img, 42, 42);
+    }
+
     //FIXME ranges are somehow arbitrary
     var weatherItems = {
         name: "Weather",
         img: atob("GBiBAf+///u5//n7//8f/9wHP8gDf/gB//AB/7AH/5AcP/AQH/DwD/uAD84AD/4AA/wAAfAAAfAAAfAAAfgAA/////+bP/+zf/+zfw=="),
         items: [
+            {
+                name: "condition",
+                get: () => ({ text: weather.txt, img: weatherIcon(weather.code),
+                  v: weather.code}),
+                show: function() { weatherItems.items[0].emit("redraw"); },
+                hide: function () {}
+            },
             {
                 name: "temperature",
                 hasRange : true,
                 get: () => ({ text: weather.temp, img: atob("GBiBAAA8AAB+AADnAADDAADDAADDAADDAADDAADbAADbAADbAADbAADbAADbAAHbgAGZgAM8wAN+wAN+wAM8wAGZgAHDgAD/AAA8AA=="),
                   v: parseInt(weather.temp), min: -30, max: 55}),
-                show: function() { weatherItems.items[0].emit("redraw"); },
+                show: function() { weatherItems.items[1].emit("redraw"); },
```

Comment:
Actually I think you can just do `this.emit("redraw");` for these. Not that it matters now, but it might make this more maintainable